src/featureFlags/dynamicConfig/quickRouteChains.ts

```diff
@@ -1,13 +1,20 @@
 import { ChainId } from '@uniswap/sdk-core'
+import { useFeatureFlagsContext } from 'featureFlags'
 
 import { DynamicConfigName, useDynamicConfig } from '.'
 
 // eslint-disable-next-line import/no-unused-modules
 export function useQuickRouteChains(): ChainId[] {
   const config = useDynamicConfig(DynamicConfigName.quickRouteChains)
-  const chains = config.get('chains', [])
+  const featureFlagsContext = useFeatureFlagsContext()
+  let chains = config.get('chains', []) as ChainId[]
+
+  const modalSetChains = featureFlagsContext.flags[DynamicConfigName.quickRouteChains]
+  if (Array.isArray(modalSetChains) && modalSetChains !== chains) {
+    chains = modalSetChains
```

Comment:
this line is for overwriting the remote values with local overrides right? is it possible that the two objects diverge so we lose some fields when assigning like this?
we could also do `chains = { ...chains, ...modalSetChains }`

Reply:
hmm, it is for overwriting so no -- if statsig chains are set to [1], but we want to test locally on [3, 10], `useQuickRouteChains()` should be [3, 10]